scripts/binaryen-lit.in

```diff
@@ -28,4 +28,7 @@ builtin_parameters = {
 }
 
 if __name__ == '__main__':
+    # Default to turning on debugging if not explicitly set
+    if 'BINARYEN_PASS_DEBUG' not in os.environ:
+        os.environ['BINARYEN_PASS_DEBUG'] = '1'
```

Comment:
Hmm, I'm not sure what this file does actually (what is prefix `.in`?)

But also I'm not sure we want to force this on all the time.

We should have already been testing this I think:

https://github.com/WebAssembly/binaryen/blob/1ec93d2ed521431ffea3152759a5b693e085d7a1/check.py#L333

That code should (but apparently isn't?) run the tests in both modes. With this PR as it is, I think it would make us run it twice in debug mode..? But maybe I'm missing something.

Reply:
Well that was a real head scratcher, but I figured out that lit runs the test commands with a (mostly) fresh environment, which is why the environment variable was not being picked up. I've rewritten the PR entirely and I'm not sure how (or whether) the previous solution worked, but this solution definitely works.

Reply:
Also, the .in file is a CMake template that gets filled out and copied to `bin/`. See https://github.com/WebAssembly/binaryen/blob/main/CMakeLists.txt#L399.

Reply:
(Also, just to clarify, `shared.with_pass_debug` only runs the given function a single time with the var set, not once with and once without)